member_card/models/partner.py

```diff
@@ -39,4 +27,5 @@ def _new_card(self, reason, user_id, barcode=False):
         }
         if barcode:
             card_data["barcode"] = barcode
-        self.env["member.card"].create(card_data)
+        card = self.env["member.card"].create(card_data)
+        self.barcode = card.barcode
```

Comment:
I am not certain I like this solution.

1. If we're going to write the barcode string on the partner, we should do it in `member.card`'s create method, not here.
2. I think we should research a way to make the compute method work.

In any case, making `member.card` readonly seems like a fairly sound idea.